src/detect-count.c

```diff
@@ -0,0 +1,83 @@
+/* Copyright (C) 2025 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#include "suricata-common.h"
+#include "rust.h"
+
+#include "detect-count.h"
+#include "detect-engine-buffer.h"
+#include "detect-engine-uint.h"
+#include "detect-parse.h"
+
+#include "util-validate.h"
+
+static void DetectCountFree(DetectEngineCtx *de_ctx, void *ptr)
+{
+    SCDetectU32Free(ptr);
+}
+
+static int DetectCountSetup(DetectEngineCtx *de_ctx, Signature *s, const char *arg)
+{
+    int sm_list = DETECT_SM_LIST_PMATCH;
+    if (s->init_data->list == DETECT_SM_LIST_NOTSET) {
+        SCLogError("count must be applied on a multi-buffer");
+        return -1;
+    }
+    if (DetectBufferGetActiveList(de_ctx, s) == -1) {
+        SCLogError("count must be applied on a multi-buffer");
+        return -1;
+    }
+
+    sm_list = s->init_data->list;
+    // TODO check it is a multi buffer and not a single buffer
```

Comment:
And add tests about impossible rules like `http.uri; count:1;` and raw tcp `content: "abc"; count: 1;`